lib/app.js

```diff
@@ -294,13 +294,20 @@ function uploadSite(settings, directoryRef, argv) {
 }
 
 function getSettings(argv) {
-  var settingsFile = path.resolve('./firebase.json');
-  var settingsJSON, settings;
-  if (!fs.existsSync(settingsFile)) {
-    console.log(chalk.red('Initialization Error') + ' - Directory not ' +
+  var curDir = process.cwd();
+  var settingsFile;
+  while(!fs.existsSync((settingsFile = (curDir + '/firebase.json')))) {
```

Comment:
also, I think the `path.resolve()` is significant in that it turns unix-style paths like `./firebase.json` and converts then to windows paths if relevant: `.\\firebase.json`